app/controllers/application_controller.rb

```diff
@@ -1,5 +1,7 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  protect_from_forgery prepend: true
```

Comment:
What is the reason for this addition? I'm sure it's useful for something, but since I don't get it today, we won't probably nether tomorrow.
I bit a explanation would be good here. :)

Furthermore:

```
protect_from_forgery with: :exception, prepend: true # ;)
```